core/src/autogluon/core/models/abstract/abstract_model.py

```diff
@@ -137,28 +137,78 @@ def __init__(self,
         self.compile_time = None  # Time taken to compile the model in seconds
         self.val_score = None  # Score with eval_metric (Validation data)
 
+        self._user_params, self._user_params_aux = self._init_user_params(params=hyperparameters)
+
         self.params = {}
         self.params_aux = {}
-
-        if hyperparameters is not None:
-            hyperparameters = hyperparameters.copy()
-        if hyperparameters is not None and AG_ARGS_FIT in hyperparameters:
-            self._user_params_aux = hyperparameters.pop(AG_ARGS_FIT)  # TODO: Delete after initialization?
-        else:
-            self._user_params_aux = None
-        if self._user_params_aux is None:
-            self._user_params_aux = dict()
-        self._user_params = hyperparameters  # TODO: Delete after initialization?
-        if self._user_params is None:
-            self._user_params = dict()
-
         self.params_trained = dict()
         self._is_initialized = False
         self._is_fit_metadata_registered = False
         self._fit_metadata = dict()
 
         self._compiler = None
 
+    @classmethod
+    def _init_user_params(cls, params: Optional[Dict[str, Any]], ag_args_fit: str = AG_ARGS_FIT, ag_arg_prefix: str = AG_ARG_PREFIX) -> (Dict[str, Any], Dict[str, Any]):
+        """
+        Given the user-specified hyperparameters, split into `params` and `params_aux`.
+
+        Parameters
+        ----------
+        params : Optional[Dict[str, Any]]
+            The model hyperparameters dictionary
+        ag_args_fit : str, default = "ag_args_fit"
+            The params key to look for that contains params_aux.
+            If the key is present, the value is used for params_aux and popped from params.
+            If no such key is found, then initialize params_aux as an empty dictionary.
+        ag_arg_prefix : str, default = "ag."
+            The key prefix to look for that indicates a parameter is intended for params_aux.
+            If None, this logic is skipped.
+            If a key starts with this prefix, it is popped from params and added to params_aux with the prefix removed.
+            For example:
+                input:  params={'ag.foo': 2, 'abc': 7}, params_aux={'bar': 3}, and ag_arg_prefix='.ag',
```

Comment:
Where do we validate the ag_args? Just wonder if adding the prefix logic would affect the validation

Reply:
We validate them after this logic. ag_args is validated during `.fit`. This logic is done during `.init`